### PR TITLE
OSDOCS#9729: Installing GitOps on MicroShift

### PR DESCRIPTION
**Pull request by kelbrown20**
**Version(s):** 4.15+ 
**Issue:** [OSDOCS-9729](https://issues.redhat.com//browse/OSDOCS-9729)

**Link to docs preview:**

- [Installing the GitOps ArgoCD manifests from an RPM package](https://72974--ocpdocs-pr.netlify.app/microshift/latest/microshift_install/microshift-install-rpm#microshift-installing-rpms-for-gitops_microshift-install-rpm)

**QE review:**
- [x] QE has approved this change.

### CODE REVIEW

File: modules/microshift-install-rpms-gitops.adoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Module included in the following assemblies:
+//
+// microshift/microshift-install-rpm.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-installing-rpms-for-gitops_{context}"]
+== Installing the GitOps ArgoCD manifests from an RPM package
+
+{microshift-short} allows the ability to use a lightweight version of the OpenShift GitOps. You can install the necessary ArgoCD manifests using a RPM package.
```
Comment (ShaunaDiaz) [inline modules/microshift-install-rpms-gitops.adoc:9]:
```suggestion
You can use a lightweight version of OpenShift GitOps with {microshift-short} to help manage your applications.  Install the ArgoCD manifests using a RPM package. 
```

File: modules/microshift-install-rpms-gitops.adoc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+{microshift-short} allows the ability to use a lightweight version of the OpenShift GitOps. You can install the necessary ArgoCD manifests using a RPM package. 
+
+.Procedure
+
+. Install the ArgoCD manifest packages by running the following command: 
+
+// I believe the rpm package is not published yet GITOPS-3790 so I don't have the command yet
+
+//I also believe reseting the manifests is the same command here
```
Comment (ShaunaDiaz) [inline modules/microshift-install-rpms-gitops.adoc:17]: Or do we need to use kustomize because we're treating it like an application?

**Comment by kelbrown20** [comment]
@anandf Can you take a look at this from the gitops side when you get the chance, thank you!

**Comment by kelbrown20** [comment]
@Srivaralakshmi Hi Sri! Was also wondering if you could take a look at this PR when you get a chance as well?

File: modules/microshift-install-rpms-gitops.adoc [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-installing-rpms-for-gitops_{context}"]
+== Installing the GitOps ArgoCD manifests from an RPM package
+
+You can use a lightweight version of the OpenShift GitOps with {microshift-short} to help manage your applications. Install the necessary ArgoCD manifests using a RPM package. 
+
+.Procedure
+
+. Install the ArgoCD manifest packages by running the following command:
```
Comment (anandf) [inline modules/microshift-install-rpms-gitops.adoc:13]: How about `Install the GitOps packages by running the following command:`

**Comment by kelbrown20** [comment]
@varshab1210 Hello do you think you could take a look at this PR from the QE side when you get the chance? Thank you!

**Review by anandf — APPROVED** [review]
LGTM

**Comment by kelbrown20** [comment]
@dhensel-rh Hi Douglas! Do you think you could take a look at this PR from the QE side when you get the chance? Thank you!

**Comment by ShaunaDiaz** [comment]
See also https://github.com/openshift/openshift-docs/pull/73201#discussion_r1537684077

File: modules/microshift-install-rpms-gitops.adoc [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+You can use a lightweight version of the OpenShift GitOps with {microshift-short} to help manage your applications. Install the necessary ArgoCD manifests using a RPM package. 
+
+.Procedure
+
+. Install the GitOps packages by running the following command: 
++
+[source,terminal]
+----
+$ sudo dnf install microshift-gitops microshift-gitops-release-info
```
Comment (ggiguash) [inline modules/microshift-install-rpms-gitops.adoc:17]:
```suggestion
$ sudo subscription-manager repos --enable=gitops-1.12-for-rhel-9-$(uname -m)-rpms
$ sudo dnf install microshift-gitops microshift-gitops-release-info
```

File: modules/microshift-install-rpms-gitops.adoc [hunk 5]
```diff
@@ -0,0 +1,25 @@
+// Module included in the following assemblies:
+//
+// microshift/microshift-install-rpm.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-installing-rpms-for-gitops_{context}"]
+== Installing the GitOps ArgoCD manifests from an RPM package
+
+You can use a lightweight version of the OpenShift GitOps with {microshift-short} to help manage your applications. Install the necessary ArgoCD manifests using a RPM package.
```
Comment (ggiguash) [inline modules/microshift-install-rpms-gitops.adoc:9]:
```suggestion
You can use a lightweight version of the OpenShift GitOps with {microshift-short} to help manage your applications. Install the necessary ArgoCD manifests using an RPM package. 
```

File: modules/microshift-install-rpms-gitops.adoc [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+.Procedure
+
+. Install the GitOps packages by running the following command: 
++
+[source,terminal]
+----
+$ sudo dnf install microshift-gitops microshift-gitops-release-info
+----
+
+. To apply the manifest from the package to an active cluster, run the following command:
```
Comment (ggiguash) [inline modules/microshift-install-rpms-gitops.adoc:20]: I think "applying manifest" is an implementation detail. What we mean to say here is that we just need to restart microshift service to complete GitOps installation.

File: modules/microshift-install-rpms-gitops.adoc [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-installing-rpms-for-gitops_{context}"]
+== Installing the GitOps ArgoCD manifests from an RPM package
+
+You can use a lightweight version of the OpenShift GitOps with {microshift-short} to help manage your applications. Install the necessary ArgoCD manifests using an RPM package. 
+
+.Procedure
+
+. Install the subscription manager by running the following command:
```
Comment (ggiguash) [inline modules/microshift-install-rpms-gitops.adoc:13]:
```suggestion
. Enable the GitOps repository by running the following subscription manager command:
```

File: modules/microshift-install-rpms-gitops.adoc [hunk 8]
```diff
@@ -0,0 +1,31 @@
+// Module included in the following assemblies:
+//
+// microshift/microshift-install-rpm.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-installing-rpms-for-gitops_{context}"]
+== Installing the GitOps ArgoCD manifests from an RPM package
+
+You can use a lightweight version of the OpenShift GitOps with {microshift-short} to help manage your applications. Install the necessary ArgoCD manifests using an RPM package.
```
Comment (ggiguash) [inline modules/microshift-install-rpms-gitops.adoc:9]: Should we explain which RPM package? Something like "... an RPM package as described in this document"?

File: modules/microshift-install-rpms-gitops.adoc [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
++
+----
+$ sudo subscription-manager repos --enable=gitops-1.12-for-rhel-9-$(uname -m)-rpms
+----
+
+. Install the GitOps packages by running the following command: 
++
+[source,terminal]
+----
+$ sudo dnf install microshift-gitops microshift-gitops-release-info
```
Comment (ggiguash) [inline modules/microshift-install-rpms-gitops.adoc:23]:
```suggestion
$ sudo dnf install -y microshift-gitops microshift-gitops-release-info
```

File: modules/microshift-install-rpms-gitops.adoc [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+----
+
+. To restart the {microshift-short} service, run the following command:
```
Comment (ggiguash) [inline modules/microshift-install-rpms-gitops.adoc:26]: We need to explain that the restart is necessary to complete the installation.
```suggestion
. To complete the installation, restart the {microshift-short} service by running the following command: 
```

File: modules/microshift-install-rpms-gitops.adoc [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
++
+----
+$ sudo subscription-manager repos --enable=gitops-1.12-for-rhel-9-$(uname -m)-rpms
+----
+
+. Install the GitOps packages by running the following command: 
++
+[source,terminal]
+----
+$ sudo dnf install -y microshift-gitops microshift-gitops-release-info
```
Comment (DanielFroehlich) [inline modules/microshift-install-rpms-gitops.adoc:23]: No need to install the release-info package. That is only required for offline/airgapped installation to determine which container images are required by gitops. Just "sudo dnf install -y microshift-gitops" is enough

File: modules/microshift-install-rpms-gitops.adoc [hunk 12]
```diff
@@ -0,0 +1,49 @@
+// Module included in the following assemblies:
+//
+// microshift/microshift-install-rpm.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-installing-rpms-for-gitops_{context}"]
+== Installing the GitOps ArgoCD manifests from an RPM package
+
+You can use a lightweight version of the OpenShift GitOps with {microshift-short} to help manage your applications. Install the necessary ArgoCD manifests using an RPM package.
```
Comment (DanielFroehlich) [inline modules/microshift-install-rpms-gitops.adoc:9]: Can we add a small paragraph on the pre-reqs here? Those are:

- MicroShift installation >=V4.14 (RPM or ostree based)
- 250 MB of additional RAM recommended, though not requested

File: modules/microshift-install-rpms-gitops.adoc (hunk 11 above)
Comment (kelbrown20) [reply modules/microshift-install-rpms-gitops.adoc:23]: Ah gotcha! Thank you!

File: modules/microshift-install-rpms-gitops.adoc (hunk 12 above)
Comment (kelbrown20) [reply modules/microshift-install-rpms-gitops.adoc:9]: Ill add a prereqs section, thank you!

File: modules/microshift-install-rpms-gitops.adoc [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+.Procedure
+
+. Enable the GitOps repository with the subscription manager by running the following command:
++
+[source,terminal]
+----
+$ sudo subscription-manager repos --enable=gitops-1.12-for-rhel-9-<architecture>-rpms
+----
+
+. Install the GitOps packages by running the following command:
```
Comment (agullon) [inline modules/microshift-install-rpms-gitops.adoc:25]:
```suggestion
. Install the GitOps package by running the following command: 
```
In this case, only 1 package is installed, `microshift-gitops-1.12.0-36.el9.x86_64.rpm`. But `dnf install` command may install multiple packages.

File: modules/microshift-install-rpms-gitops.adoc [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+----
+
+. Install the GitOps packages by running the following command: 
++
+[source,terminal]
+----
+$ sudo dnf install -y microshift-gitops 
+----
+
+. To complete the installation, restart the {microshift-short} service by running the following command:
```
Comment (agullon) [inline modules/microshift-install-rpms-gitops.adoc:32]:
```suggestion
. To deploy Argo CD pods, restart the {microshift-short} service by running the following command: 
```
From my point of view, `microshift-gitops` is already installed in the previous step, but by restarting `microshift` we can start Argo CD pods.

Sometimes we all use `installation` as a general concept that means something is ready to be used, so not sure what's the right wording for this step.

File: modules/microshift-install-rpms-gitops.adoc [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+* You installed {microshift-short} version 4.14 or higher
+* Additional RAM storage of 250MB recommended 
+
+.Procedure
+
+. Enable the GitOps repository with the subscription manager by running the following command:
++
+[source,terminal]
+----
+$ sudo subscription-manager repos --enable=gitops-1.12-for-rhel-9-<architecture>-rpms
```
Comment (agullon) [inline modules/microshift-install-rpms-gitops.adoc:22]:
```suggestion
$ sudo subscription-manager repos --enable=gitops-1.12-for-rhel-9-$(uname -m)-rpms
```

**Comment by agullon** [comment]
hi @kelbrown20 not sure if this is important, but I just realize that it is not installing Argo CD CLI, only gitops underlying functionality. So maybe it's worth to mention that Argo CD CLI is not available.

**Comment by kelbrown20** [comment]
> hi @kelbrown20 not sure if this is important, but I just realize that it is not installing Argo CD CLI, only gitops underlying functionality. So maybe it's worth to mention that Argo CD CLI is not available.

That is a great point! I'll add a note with that, thank you

**Review by agullon — APPROVED** [review]
/lgtm

**Review by cbippley — COMMENTED** [review]
Great job, LGTM!

**Pull request merged**